### PR TITLE
feat: add marks handler

### PR DESCRIPTION
**Pull request by smoka7**

### CODE REVIEW

**Comment by lewis6991** [comment]
Similar to how we handle folds (https://github.com/lewis6991/satellite.nvim/blob/d79f99810e09ca2e1f27c49d066655a5d1fe5654/lua/satellite.lua#L912-L920), we will need to refresh everytime a mark is added or deleted.

File: lua/satellite/handlers/marks.lua [hunk 1]
```diff
@@ -0,0 +1,25 @@
+local highlight = 'Normal'
+
+---@type Handler
+local handler = {
+	name = 'marks',
```
Comment (lewis6991) [inline lua/satellite/handlers/marks.lua:5]: Convert tabs to spaces

**Review by lewis6991 — COMMENTED** [review]
Also update the README

**Comment by smoka7** [comment]
Take a another look. I think it's ready

File: lua/satellite/handlers/marks.lua [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    vim.keymap.set({ 'n', 'v' }, map, function()
+      vim.schedule(refresh)
+      return map
+    end, { unique = true, expr = true })
+  end
+end
+
+function handler.update(bufnr)
+  local marks = {}
+  local buffer_marks = vim.fn.getmarklist(bufnr)
```
[the remark this answers is not in the document]
Comment (lewis6991) [reply lua/satellite/handlers/marks.lua:25]: I think for now, maybe just display user marks (`a-z`). The other marks can be enabled with an option.

[the remark this answers is not in the document]
Comment (smoka7) [reply lua/satellite/handlers/marks.lua:25]: ok i change it.

File: lua/satellite/handlers.lua [hunk 3]
```diff
@@ -3,7 +3,7 @@
 ---@field name string
 ---@field ns integer
 ---@field init fun()
----@field update fun(bufnr: integer)
+---@field update fun(bufnr: integer,user_config: table)
```
Comment (lewis6991) [inline lua/satellite/handlers.lua:6]: Config should be passed to `init()`

**Review by lewis6991 — CHANGES_REQUESTED** [review]

File: lua/satellite/handlers.lua (hunk 3 above)
Comment (smoka7) [reply lua/satellite/handlers.lua:6]: sorry i couldn't figure out where to pass it

Comment (lewis6991) [reply lua/satellite/handlers.lua:6]: It's ok. I'm still figuring a lot of this out. It still might change.

**Pull request merged**